aerosonicdb/features/extract_env_mfccs.py

```diff
@@ -24,91 +30,105 @@
 EXPECTED_MFCC_VECTORS_PER_SEGMENT = math.ceil(SAMPLES_PER_SEGMENT / HOP_LENGTH)
 
 
-def save_env_mfccs(env_n,
-                   duration=DURATION,
-                   n_mfcc=N_MFCC,
-                   n_fft=N_FFT,
-                   sample_rate=SAMPLE_RATE,
-                   hop_length=HOP_LENGTH,
-                   class_path=CLASS_PATH,
-                   output_path=OUTPUT_PATH,
-                   env_audio_path=ENV_AUDIO_PATH,
-                   ignore=True):
-
-    audio_path = os.path.join(env_audio_path, f'{str(env_n)}_AUDIO.wav')
-    json_path = os.path.join(output_path, f'{str(env_n)}_ENV_{n_mfcc}_mfcc_{duration}.json')
+def save_env_mfccs(
+    env_n,
+    duration=DURATION,
+    n_mfcc=N_MFCC,
+    n_fft=N_FFT,
+    sample_rate=SAMPLE_RATE,
+    hop_length=HOP_LENGTH,
+    class_path=CLASS_PATH,
+    output_path=OUTPUT_PATH,
+    env_audio_path=ENV_AUDIO_PATH,
+    ignore=True,
+    force=False,
+):
+
+    audio_path = os.path.join(env_audio_path, f"{str(env_n)}_AUDIO.wav")
+    json_path = os.path.join(
+        output_path, f"{str(env_n)}_ENV_{n_mfcc}_mfcc_{duration}.json"
+    )
     class_map = pd.read_csv(class_path)
 
     if not os.path.exists(audio_path):
-        print('Environment audio not found')
-
-    if not os.path.exists(json_path):
-
-        data = {
-            'mfcc': [],
-            'class_label': []
-        }
-
-        signal, sr = librosa.load(audio_path)
-
-        if sr == sample_rate:
-
-            clip_segments = len(signal) // SAMPLES_PER_SEGMENT
-
-            for s in range(clip_segments):
-                start = SAMPLES_PER_SEGMENT * s
-                end = start + SAMPLES_PER_SEGMENT
-
-                mfcc = librosa.feature.mfcc(y=signal[start:end],
-                                            sr=sample_rate,
-                                            n_mfcc=n_mfcc,
-                                            n_fft=n_fft,
-                                            hop_length=hop_length)
-                mfcc = mfcc.T
-
-                n_str = str(env_n - 1)
-
-                class_label = class_map[n_str].iloc[s]
-                
-                if ignore:
-                    
-                    # logic to skip "edge" cases
-                    if class_label == 'ignore':
-                        pass
+        print("Environment audio not found")
+        return
+
+    if os.path.exists(json_path) and not force:
+        print(
+            f"JSON feature descriptor file already exists for environment #{env_n} - not generating features."
+        )
+        return
+
+    data = {"mfcc": [], "class_label": []}
+
+    signal, sr = librosa.load(audio_path)
+
+    if sr == sample_rate:
+        clip_segments = len(signal) // SAMPLES_PER_SEGMENT
+
+        for s in tqdm(range(clip_segments)):
```

Comment:
wrapping in `tqdm` gives us a nice progress bar, works both for CLI and notebook invocation